src/main/java/it/gov/pagopa/pu/debtpositions/service/statusalign/DebtPositionHierarchyStatusAlignerServiceImpl.java

```diff
@@ -75,4 +76,32 @@ public DebtPositionDTO finalizeSyncStatus(Long debtPositionId, Map<String, IupdS
 
     return debtPositionMapper.mapToDto(debtPosition);
   }
+
+  @Override
+  public DebtPositionDTO notifyReportedTransferId(Long transferId) {
+    DebtPosition debtPosition = debtPositionRepository.findByTransferId(transferId);
+
+    debtPosition.getPaymentOptions().forEach(paymentOption ->
+      paymentOption.getInstallments().stream()
+        .filter(installment -> {
+          boolean isToBeReported = installment.getTransfers().stream().anyMatch(transfer -> transfer.getTransferId().equals(transferId));
+          if (isToBeReported && !installment.getStatus().equals(InstallmentStatus.PAID)) {
+            throw new InvalidInstallmentStatusException("The installment is not in the paid status to be set in reported status");
+          }
+          return isToBeReported;
```

Comment:
let's handle the useCase where, the status could already be REPORTED.
In this case let's just return false in order to skip the update on the DB, but let's not throw the exception